extension.js

```diff
@@ -309,6 +318,8 @@ function track_window(win) {
     set_window_above();
     set_window_stick();
     set_skip_taskbar();
+    if (settings.get_boolean('window-maximize') && !win.maximized_vertically)
+        win.maximize(Meta.MaximizeFlags.VERTICAL);
```

Comment:
I'm not 100% sure, but maybe you can simply call `set_window_maximize()` here now?

Reply:
Yep, `set_window_maximize()` seems to work fine here